src/OpenColorIO/ops/fixedfunction/FixedFunctionOpGPU.cpp

```diff
@@ -347,6 +349,1070 @@ void Add_GamutComp_13_Inv_Shader(GpuShaderText & ss,
+    ss.newLine() << "lms = " << "lms / " << ss.float3Const(p.D_RGB[0], p.D_RGB[1], p.D_RGB[2]) << ";";
+
+    ss.newLine() << pxl << ".rgb = " << ss.mat3fMul(&p.MATRIX_CAM16_to_RGB[0], "lms") << ";";
+}
+
+std::string _Add_Reach_table(
+    GpuShaderCreatorRcPtr & shaderCreator,
+    unsigned resourceIndex,
+    const ACES2::Table1D & table)
+{
+    // Reserve name
+    std::ostringstream resName;
+    resName << shaderCreator->getResourcePrefix()
+            << std::string("_")
+            << std::string("reach_m_table")
+            << resourceIndex;
+
+    // Note: Remove potentially problematic double underscores from GLSL resource names.
+    std::string name(resName.str());
+    StringUtils::ReplaceInPlace(name, "__", "_");
+
+    // Register texture
+    GpuShaderDesc::TextureDimensions dimensions = GpuShaderDesc::TEXTURE_1D;
+    if (shaderCreator->getLanguage() == GPU_LANGUAGE_GLSL_ES_1_0
+        || shaderCreator->getLanguage() == GPU_LANGUAGE_GLSL_ES_3_0
+        || !shaderCreator->getAllowTexture1D())
+    {
+        dimensions = GpuShaderDesc::TEXTURE_2D;
+    }
+
+    shaderCreator->addTexture(
+        name.c_str(),
+        GpuShaderText::getSamplerName(name).c_str(),
+        table.size,
+        1,
+        GpuShaderCreator::TEXTURE_RED_CHANNEL,
+        dimensions,
+        INTERP_NEAREST,
+        &(table.table[0]));
+
+
+    if (dimensions == GpuShaderDesc::TEXTURE_1D)
+    {
+        GpuShaderText ss(shaderCreator->getLanguage());
+        ss.declareTex1D(name);
+        shaderCreator->addToDeclareShaderCode(ss.string().c_str());
+    }
+    else
+    {
+        GpuShaderText ss(shaderCreator->getLanguage());
+        ss.declareTex2D(name);
+        shaderCreator->addToDeclareShaderCode(ss.string().c_str());
+    }
+
+    // Sampler function
+    GpuShaderText ss(shaderCreator->getLanguage());
+
+    ss.newLine() << ss.floatKeyword() << " " << name << "_sample(float h)";
+    ss.newLine() << "{";
+    ss.indent();
+
+    ss.newLine() << ss.floatDecl("hwrap") << " = h;";
+    ss.newLine() << "hwrap = hwrap - floor(hwrap / 360.f) * 360.f;";
+    ss.newLine() << "hwrap = (hwrap < 0.0f) ? hwrap + 360.f : hwrap;";
+
+    ss.newLine() << ss.floatDecl("i_lo") << " = floor(hwrap);";
+    ss.newLine() << ss.floatDecl("i_hi") << " = (i_lo + 1);";
+    ss.newLine() << "i_hi = i_hi - floor(i_hi / 360.f) * 360.f;";
+
+    if (dimensions == GpuShaderDesc::TEXTURE_1D)
+    {
+        ss.newLine() << ss.floatDecl("lo") << " = " << ss.sampleTex1D(name, "(i_lo + 0.5f) / 360.f") << ".r;";
+        ss.newLine() << ss.floatDecl("hi") << " = " << ss.sampleTex1D(name, "(i_hi + 0.5f) / 360.f") << ".r;";
+    }
+    else
+    {
+        ss.newLine() << ss.floatDecl("lo") << " = " << ss.sampleTex2D(name, "vec2((i_lo + 0.5f) / 360.f, 0.5f)") << ".r;";
+        ss.newLine() << ss.floatDecl("hi") << " = " << ss.sampleTex2D(name, "vec2((i_hi + 0.5f) / 360.f, 0.5f)") << ".r;";
```

Comment:
Sorry I left a previously unpublished review comment here: `float2Const` should be used here instead of `vec2` (same elsewhere).

Reply:
Thanks @num3ric good catch, I fixed this in the latest commit.